### PR TITLE
feat: envWithDefault cheatcode initial implementation #3732

### PR DESCRIPTION
**Pull request by slvDev**
Implementation of Issue #3732

That is my first PR to Foundry - please correct me if I got anything wrong.

## Motivation

As noted by @vicnaum, there is a demand for non-reverting cheatcode for reading from Environment

## Solution

Based on @mds1 suggestion: a new cheatcode was created named `vm.envWithDefault(string name, <defaultVal>)`, where <defaultVal> type determines the type and encoding of the Environment Variable

### CODE REVIEW

**Review by gakonst — APPROVED** [review]
beautiful - pending CI fixes

File: evm/src/executor/abi/mod.rs [hunk 1]
```diff
@@ -41,6 +41,20 @@ ethers::contract::abigen!(
             envBytes32(string,string)(bytes32[])
             envString(string,string)(string[])
             envBytes(string,string)(bytes[])
+            envWithDefault(string,bool)(bool)
```
Comment (mattsse) [inline evm/src/executor/abi/mod.rs:44]: I'd like to suggest an alternative name here:

`envOr` or even variants for all types like `envStringOr`

imo `Default` is not really fitting here

Comment (mds1) [reply evm/src/executor/abi/mod.rs:44]: I'm fine with `envOr`, just originally chose `envWithDefault` cause it felt more explicit, though `envOr` is more aligned with rust-like syntax

**Comment by gakonst** [comment]
`envOr` :+1:

**Comment by gakonst** [comment]
full send - thanks @slvDev

**Pull request merged**